config/config.go

```diff
@@ -1850,9 +1850,14 @@ func (c *Config) UpdateConfig(configPath string, newCfg *Config, dryrun bool) er
 	return c.LoadConfig(configPath, dryrun)
 }
 
-// GetConfig returns a pointer to a configuration object
+// GetConfig returns the global shared config instance
 func GetConfig() *Config {
-	return &Cfg
+	return &cfg
+}
+
+// SetConfig sets the global shared config instance
+func SetConfig(c *Config) {
+	cfg = *c
```

Comment:
In its current use, there is no need to use the mutex `m`. I can foresee this being a race at some other time and suggest using `m` to protect from that like it protects elsewhere in this package. I'm not preventing merge since its pre-existing

Reply:
I'll do that. Trying not to get too into it because I'm sure this is a 🧵 I'll keep pulling at...

Fixed 26d22b18

Avoided doing RWMutex and fixing other uses at the same time.